### DISCUSSION TITLE
Warning: Extra attributes from the server: data-lt-installed

### DISCUSSION THREAD

**Discussion by itsezc** [vercel/next.js#41816] [App Router]
For anyone running into this warning on Next 13 (on Firefox):

```
Warning: Extra attributes from the server: data-lt-installed
html
ReactDevOverlay@webpack-internal:///../../node_modules/next/dist/client/components/react-dev-overlay/internal/ReactDevOverlay.js:53:9
HotReload@webpack-internal:///../../node_modules/next/dist/client/components/react-dev-overlay/hot-reloader-client.js:19:39
Router@webpack-internal:///../../node_modules/next/dist/client/components/app-router.js:73:89
ErrorBoundaryHandler@webpack-internal:///../../node_modules/next/dist/client/components/error-boundary.js:28:9
ErrorBoundary@webpack-internal:///../../node_modules/next/dist/client/components/error-boundary.js:40:42
AppRouter
ServerRoot@webpack-internal:///../../node_modules/next/dist/client/app-index.js:114:25
RSCComponent
Root@webpack-internal:///../../node_modules/next/dist/client/app-index.js:130:25
```

You have to disable / uninstall the LanguageTool extension, disabling it for the site won't be enough.

**Comment by oscartbeaumont** [comment]
I was getting something like this (`react_devtools_backend.js:4026 Warning: Extra attributes from the server: data-new-gr-c-s-check-loaded,data-gr-ext-installed
    at body`) and it was caused by me having [Grammarly](https://www.grammarly.com/browser/chrome) installed. Maybe you have a browser extension which is adding those attributes?

**Reply by itsezc** [reply]
Yes, I have an extension called LanguageTool similar to Grammarly that causes the issue

**Reply by oscartbeaumont** [reply]
Sorry missed that in the initial post, my bad.

**Reply by prabhunath-code** [reply]
thankyou soo much you are a legend man

**Reply by valourus** [reply]
For me `language tool` had the same effect, thanks for saving me hours of work :)

**Reply by bishoe01** [reply]
Thank you so much. It worked out great.

**Reply by monirulislamkhan** [reply]
Thank you. now its work.

**Reply by such-xl** [reply]
Yes, bro.

**Reply by asumaran** [reply]
it was Grammarly for me 🤦

**Reply by Voltzz-stv** [reply]
Grammarly for me too🤯. Thanks @oscartbeaumont .

**Reply by mateusarcedev** [reply]
Yes to me. Thanks!

**Reply by xneyk** [reply]
It worked for me too! I think in my case was LanguageTool causing the warning, it makes sense because both are grammar extensions, maybe both uses some same dependencies.

**Reply by pratul03** [reply]
same issue cuz of grammarly

**Reply by MobinaYousefian** [reply]
same issue with "AI Grammar Checker & Paraphraser – LanguageTool".
Thank you so much. Absolutely clever idea!

**Comment by Undead34** [comment]
For development, it's better to have everything disabled, even AdBlock, how did you find out about LanguageTool? Very clever if you ask me, I would have never thought of it.

**Reply by Bilalzip** [reply]
it was Grammarly

**Comment by Tech-Code1** [comment]
I have not been able to solve it in any way, I uninstalled all the extensions and still I get the error, what is it due to?

![Pantallazo 19-01-2023 00 27 42](https://user-images.githubusercontent.com/61479618/213362861-6714c65d-3bde-4edb-97f2-5f3e23cc75fc.png)

**Reply by Tech-Code1** [reply]
This way I have my layout in the root
![Pantallazo 19-01-2023 00 30 00](https://user-images.githubusercontent.com/61479618/213363053-669bb690-79b0-4f4a-b8e2-c9c67acc1244.png)

**Reply by Ahmed-Sbai** [reply]
how did you manage that?

**Comment by Spale96** [comment]
Had same error, but once I disabled Grammar extension, it disappears.

**Reply by phtn** [reply]
The culprit was Grammarly on mine. Switched "This can read and change site data" -> When you click the extension

**Reply by AdehenryOmoOba** [reply]
Thanks, i disabled it and the error disappeared

**Reply by Sana-Mustafa** [reply]
yes it truly work when I also remove Grammarly extension from the browser

**Comment by donatoaguirre24** [comment]
Hi @itsezc 👋🏻 

Does this cause any harm besides the warning? Asking because I would rather not uninstall LanguageTool.

**Reply by itsezc** [reply]
Just a warning, never run into anything more serious.

**Reply by donatoaguirre24** [reply]
Thanks!

**Reply by amoweolubusayo** [reply]
Thanks. Just ran into same error. Will ignore as I just need it for a demo

**Comment by alexkates** [comment]
Just ran into this as well. Confirming that disabling Grammarly resolves the warning. I'm curious what is going on here though ...

**Comment by danspratling** [comment]
I've run into this problem with a few of my extensions. Grammarly, 1Password and even `next/font` seem to cause this warning.

Seems like it's not really a big problem though, but I would like to get rid of it without losing the functionality of any of my tools (I obviously don't want myself or others to be trained to ignore errors in the console).

**Reply by danspratling** [reply]
FYI, apparently you can disable warnings if you know what's causing them by doing the following

```
//app/layout.tsx

export default RootLayout({ children }) {
  return (
    <html lang="en" className={customFont.variable}>
      <body
          suppressHydrationWarning={true} // This prevents extensions from causing a server/client mismatch
        >
          {children}
        </body>
      </html>
  )
}

**Reply by ETOPS7** [reply]
I don't want to delete Grammarly, if it's the main reason, actually

`suppressHydrationWarning={true}` solved the problem with `ata-new-gr-c-s-check-loaded,d` , but the warning with `ta-gr-ext-installed` is still left

**Reply by msaouab** [reply]
thank you it's help me to solve my problem

**Comment by mdshohidulislam1486** [comment]
Nothing is working for me

**Reply by luanfvieira** [reply]
Add suppressHydrationWarning={true} in html tag.
```html
    <html lang="en" suppressHydrationWarning={true}>
```

**Reply by ghost** [reply]
This fixed it! thx

**Reply by rattanakneariroth** [reply]
Thanks this fixed it

**Reply by aidysp** [reply]
Thanks

**Reply by OmarDugha03** [reply]
This fixes it thanks

**Reply by OmarDugha03** [reply]
This fixes it thanks

**Reply by katzohub** [reply]
Thanks

**Comment by wilfred-py** [comment]
Disabling Grammarly worked for me too. Thanks!

**Comment by KajiyamaVK** [comment]
Apparently there are a few extensions that might be causing this. For me, I had to run my browser in private mode. (I had some extensions configured to run even in private, so I had to turn this option off on Firefox).

**Comment by Jacobkyalo** [comment]
I disabled Grammarly extension and it worked for me

**Comment by rakibulinux** [comment]
Stop `ColorZilla `and `Grammarly `extensions fix my problem.

![image](https://github.com/vercel/next.js/assets/59226530/d50d9987-bd31-4567-b1cc-f183f4b1b125)

**Comment by im-nasirr** [comment]
Thanks rakibulinux

**Comment by yaroslavert** [comment]
Chrome has released a new feature to disable extensions for specific URLs which is great, since this warning seems to just be a development environment issue. Simply put chrome://flags/#extensions-menu-access-control into your Chrome browser to enable this flag, then you can toggle off your extensions for the http://localhost:3000 URL! No need for the suppressHydrationWarning={true} prop mentioned above.

**Reply by sovetski** [reply]
Can you give more informations on how to do it please? What should we do here?

![image](https://github.com/vercel/next.js/assets/13520683/d10a6311-072d-439d-b6f2-521317e9568f)

**Reply by Undead34** [reply]
Cool 😎

**Reply by raychz** [reply]
Change the "Extensions Menu Access Control" setting to "Enabled", then relaunch Chrome.  You'll see a new option in the toolbar that allows you to block extensions on a per-site basis.

**Reply by sovetski** [reply]
thank you, It looks like it's not working very well, it's buggy for me but maybe over time they will fix it

**Reply by manir22sust** [reply]
Disabling Grammarly worked for me too. Thanks!

**Comment by felipealvessi** [comment]
PT:
No meu caso aconteceuu por conta do LanguageTool, ColorZilla  e Grammarly. Depois que desativei deu certo.

EN:

In my case, it happened because of LanguageTool, ColorZilla and Grammarly. After I deactivated it, it worked.

**Comment by hshoja** [comment]
```
Warning: Extra attributes from the server: data-be-installed
    at html
    at RedirectErrorBoundary
```

Disabling [Liner](https://chrome.google.com/webstore/detail/liner-chatgpt-ai-copilot/bmhcbmnbenmcecpmpepghooflbehcack) worked for me. Thanks!

**Comment by js-talentelgia** [comment]
Work for me, Disable the extension and enable it again and the warning will be removed

**Comment by ScreamZ** [comment]
Nice troll from extensions, same issue

**Comment by Marcel-Ng** [comment]
So as pointed out in the previous comments above extensions similar to Grammarly, and ColorZilla can be the culprit. In my case I had the two. These extensions passing extra attributes with your code when it is executed on the browser trying to interact with the UI, this creates a mismatch between what was rendered on the server and what is rendered on the client. While disabling these extension can solve the problem an `in code` way to solve this error is by suppressing hydration warning like so:

- In the RootLayout or layout file

export default RootLayout({ children }) {
  return (
   <html lang="en">
      <body suppressHydrationWarning={true}>
        {children}
      </body>
    </html>
  )
}

**Comment by vigneshwaran-48** [comment]
For me changing my layout component's name to **RootLayout** from **layout** solves the problem

**Reply by ScreamZ** [reply]
Good game, you disabled your NextJS custom layout.

This cannot be done. layout.tsx is a required name for layout system

**Comment by masanosukesan** [comment]
For me, the ClickUp extension happened to cause the issue as well

**Comment by bishoe01** [comment]
Grammarly for me too  😭

**Comment by ArnaudBand** [comment]
Is there any way to remove this error without disabling extensions? Any piece of code?

**Reply by misterioso013** [reply]
Use your browser's incognito tab.
